### PR TITLE
New Space Ruin: Space Mall

### PR DESCRIPTION
**Pull request by goober3**
Adds a Space Mall ruin (for space), an independent space shopping centre (infested with spiders).

Featuring a Fuckton™ of (notable) loot: -9mm ammo box + x3 9mm pistol magazines 
-x2 commander pistols
-x2 disablers
-Double-barreled shotgun + rubbershots
-Segway
-Condimaster
-Hi-tech janitor appliances
-<3500 credits
-Jukebox
-x1 Advanced medical kit (hidden) + x3 Toxin kits + x2 Regular first aid kits
-Bluespace bodybag
-Bluespace RPED
-Randomly assorted rare stock parts
-Supercooled water vapor canister (173.15 kelvin)
-x2 Thermomachines
-A cheap phrasebook for visiting Kalixcians
-Autolathe
-Service techfab
-x2 jaws of life + experimental welder + hand drill
-x1 Comically Large Crowbar
-x2 Superpacmans (+25 uranium)

## All the SCREENSHOTS

in

ru

The Ruin

- [yes!!!] I affirm that I have tested all of my proposed changes and that any issues found during tested have been addressed.

## Why It's Good For The Game

More Ruins ! ! !

## Changelog

:cl:
add: New space ruin: A derelict space mall infested with spiders.
/:cl:

### CODE REVIEW

**Comment by ZephyrTFA** [comment]
that's a lot of loot, for n relatively little danger

**Comment by Zevotech** [comment]
add a giant enemy spider

**Comment by goober3** [comment]
good point. i'll add more eggs, they spawn a fuckton of spiderlings. and maybe remove the mining vendor.

**Comment by Zevotech** [comment]
make sure you dont use the ghostrole spider eggs

**Comment by Sun-Soaked** [comment]
> make sure you dont use the ghostrole spider eggs

honestly? one or two of these might be interesting, we were discussing the possibility of having more player based defenses in ruins a while ago

might lead to funny rp stuff too

**Comment by goober3** [comment]
guh. do we even have ghostrole spider eggs here? cause i'd totally add them if we did

**Comment by Sun-Soaked** [comment]
> guh. do we even have ghostrole spider eggs here? cause i'd totally add them if we did

my intuition says definitely but they might be antag-coded due to being connected to the old spider midround antag

**Comment by PiperDoots** [comment]
The one thing I feel this could really benefit from is ghost roles. Would fit well within the concept and could bring a lot more life to it.

**Comment by Valorium-RB** [comment]
At risk of advocating for bloat, it could be neat if there were two variants of this ruin - one infested with spiders and more run-down, and another which is clean, new and staffed with several ghost roles.  Failing that, spiderling spawners would be fun!

**Comment by goober3** [comment]
I will consider a Clean™ version when I figure out ghost roles and blacklists (and if the maptainers approve). thank you for the suggestion

File: code/game/area/areas/ruins/space.dm [hunk 1]
```diff
@@ -549,3 +549,29 @@
...
+
+/area/ruin/space/has_grav/spacemall/shop2
+	name = "Space Mall Shop"
+	icon_state = "bluenew"
+
+/area/ruin/space/has_grav/spacemall/maint
+	name = "Space Mall Maintenance"
+	icon_state = "yellow"
+
+/area/ruin/space/has_grav/spacemall/dorms
```
Comment (Erikafox) [inline code/game/area/areas/ruins/space.dm:585]: could you fold this area into the main one?

**Review by Erikafox — CHANGES_REQUESTED** [review]
StrongDMM_JtkYQJCT6T
This area needs some visible damage, more webs, generally to be junked up more. 
StrongDMM_oJ5Hyg8RnX
Ditto for here
StrongDMM_G64fqjjnTe
just the damage here.
StrongDMM_eYAkYrV4FY
Get a real tile under the SSU. I recommend tech/grid or whatever you feel fits there.
StrongDMM_o5UFXHv5sB
I recommend putting corpse spawners in here, along with some blood, and spiders. Junk it up if you do that. 
image
I would put a spider in here too

You should seriously consider swapping out airlocks for airlock assemblies in any area the spiders are supposed to control. Damage helps convey danger. 

StrongDMM_pQSctMQZBg
This room needs decal work. 

I'd mark the exact line for these but the diff isn't loading. 
StrongDMM_TbAEXdbIiK
They won't work well. Use one of the subtypes instead.

StrongDMM_AhtA4OJY9y
I would not put a magical SMES here. They're best hidden away to power whole ruins instead. Swap it out for a normal SMES and varedit it so it starts with a full charge.


I'm gonna load it locally and see if there's anything that catches my eye there.

File: code/game/area/areas/ruins/space.dm (hunk 1 above)
Comment (goober3) [reply code/game/area/areas/ruins/space.dm:585]: wdym?

**Comment by Erikafox** [comment]
dreamseeker_nRE0dCPCOT
Use a visible pipe here so it's not hidden by the floor.

dreamseeker_jfy8kIni4M
The bathrooms are devoid of light. 

image
This area just feels fairly empty. What I'd recommend doing is having the "Lounge" area immediately accessible, putting more things in it, and then having the rest of the dorms behind that. Maybe give it a check in counter and make it a space hotel. 

image
You have space tiles under these windows here. 

I'd also recommend adding some girders to maint.

**Comment by goober3** [comment]
Oh God Wait I Forgot The Firelocks

File: code/game/area/areas/ruins/space.dm (hunk 1 above)
Comment (Erikafox) [reply code/game/area/areas/ruins/space.dm:585]: Either expand the dorms enough contentwise to need their own area def, or remove the area def and have the dorms be under the /spacemall generic one.

**Comment by goober3** [comment]
thank you for the review, will be reviewing your review. and fixing the Ruin. thank you

**Comment by Erikafox** [comment]
image
It seems like the spiders also will eat Bass Pro. Maybe use tinted windows for that store.

**Comment by goober3** [comment]
> image 
It seems like the spiders also will eat Bass Pro. Maybe use tinted windows for that store.

I can't believe they would do that

**Comment by Latentish** [comment]
> At risk of advocating for bloat, it could be neat if there were two variants of this ruin - one infested with spiders and more run-down, and another which is clean, new and staffed with several ghost roles. Failing that, spiderling spawners would be fun!

I really don't think we should have ruin alts unless the alts are like, phenomenal

**Comment by goober3** [comment]
Oops My Areas

**Comment by goober3** [comment]
Okay Nevermind. everything is fine. this merge conflict has been FIXED.

**Review by Erikafox — APPROVED** [review]

**Comment by ArdentSpark** [comment]
I found something strange with spider behaviour, in that they really really really do not like stepping into or sometimes attacking tiles with windoors or railings. this has a notable effect on one area in particular 
image

The spiders walk up to it, get confused and flee without attacking the windoors.

It can be gamed pretty hard! Just raising in case this is something you'd want edited, as it can be used to flee very effectively from the engineering horde of spiders

**Pull request merged**